### PR TITLE
Show student name within instructor assessment dashboards

### PR DESCRIPTION
**Pull request by nicolasruscher**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Currently within the exercise specific assessment dashboards for file upload exercises and text exercises and programming exercises no student names are shown. In this PR we make the handling uniform across all exercise types.

### Description

Student names are now shown for file upload exercises and text exercises and programming exercises within the specific assessment dashboards. 

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration -> exercises -> submissions
3. you should see students names 

### Screenshots

image
image
image
image

### CODE REVIEW

**Review by tiffca — APPROVED** [review]
Tested successfully

**Review by airelawaleria — APPROVED** [review]

**Review by EneaGore — DISMISSED** [review]

**Review by JannisBittlmayer — DISMISSED** [review]
Works as intended

**Review by FrankeLukas — APPROVED** [review]
Names are missing for programming exercises, but that is not in the scope of the PR apparently.

File: src/main/webapp/i18n/de/exercise.json [hunk 1]
```diff
@@ -105,7 +105,7 @@
             "personalRepositoryClone": "Dein persönliches Repository wurde eingerichtet. Klicke auf den Button <i>Repository klonen</i>!",
             "personalRepositoryOnline": "Dein persönliches Repository wurde eingerichtet. Klicke auf den Button <i>Programmiereditor öffnen</i>!",
             "startError": "<strong>Uh oh! Etwas ist schief gelaufen... Bitte versuch es in wenigen Minuten noch einmal die Aufgabe zu starten.</strong>",
-            "studentName": "Name",
+            "studentName": "Studenten Name",
```
Comment (SimonEntholzer) [inline src/main/webapp/i18n/de/exercise.json:108]:
```suggestion
            "studentName": "Studentenname",
```

**Review by SimonEntholzer — CHANGES_REQUESTED** [review]
Except one small change everything is looking good. Testing & code

**Review by FlorianStang — DISMISSED** [review]
worked as described.

**Review by iozbeyli — DISMISSED** [review]
Code looks good 👍

**Review by JohannesWeiss — APPROVED** [review]
Code looks good to me

**Review by FuchsDominik — APPROVED** [review]
Code looks fine

**Review by tiffca — APPROVED** [review]
re-approve

**Review by airelawaleria — APPROVED** [review]
re-approve

**Review by SimonEntholzer — APPROVED** [review]
tested again, now the name is shown for all exercise types :+1:

**Review by simonzilker — DISMISSED** [review]
works as described

**Review by JohannesWeiss — DISMISSED** [review]
Re-Approve the changes :)

**Review by marvinchimeka — APPROVED** [review]
Tested on ts2. 👍

**Review by krusche — CHANGES_REQUESTED** [review]
* The name is now shown twice for file upload
* `*ngIf="exercise.isAtLeastInstructor"` is missing for the columns (header + rows)
* For modeling the header column needs to renamed to "Student name" as well.

==> In all tables, the column should be shown once as the second column (after the numbering column #)

**Review by philgabr — DISMISSED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**